ipa-core/src/protocol/ipa_prf/malicious_security/prover.rs

```diff
@@ -171,9 +171,9 @@ impl<F: PrimeField, const L: usize, const P: usize, const M: usize> ProofGenerat
                 let (u_chunk, v_chunk) = polynomial.borrow();
                 (
                     // new u value
-                    lagrange_table_r.eval(u_chunk)[0],
+                    lagrange_table_r.eval_first(u_chunk),
```

Comment:
this table has only one row 

```rust
let lagrange_table_r = LagrangeTable::<F, L, 1>::new(&denominator, &r);
```

I am not sure if we actually do throwaway work here

Reply:
ohh, that's why this isn't speeding up at all. Gonna just pretend I didn't write this one XD